### PR TITLE
Fix chat + local feedback in mic and add remote voice

### PR DESCRIPTION
**Pull request by lalalune**
This PR does the following:
1. Mutes the local player so there isn't feedback into their own system
2. Separates the microphone and audio inputs -- all remote players have an audio input worker and local player has both an audio input worker (for the AI voice etc) and a microphone worker (for microphone to send over the network and to mouth but muted)

This PR enables multiplayer voice but that will come after this and https://github.com/webaverse/app/pull/3207 have been pulled in.

### CODE REVIEW

File: avatars/microphone-worklet.js [hunk 1]
```diff
@@ -107,10 +106,14 @@ class VolumeProcessor extends AudioWorkletProcessor {
           data: mergedSamples,
         }, [mergedSamples.buffer]);
       };
-      emitBuffer && _emitBuffer();
+      this.emitBuffer && _emitBuffer();
     }
 
     return true;
+  } catch (error){
+    console.error(error);
```
Comment (avaer) [inline avatars/microphone-worklet.js:114]: Is this try-catch required for correctness?

File: avatars/microphone-worklet.js [hunk 2]
```diff
@@ -1,63 +1,62 @@
...
         if (args.emitBuffer !== undefined) {
-          emitBuffer = args.emitBuffer;
+          this.emitBuffer = args.emitBuffer;
         }
       }
     });
     this.port.start();
   }
   process(inputs, outputs) {
+    try {
```
Comment (lalalune) [inline avatars/microphone-worklet.js:34]: remove try?

File: character-controller.js [hunk 3]
```diff
@@ -259,16 +272,33 @@ class PlayerBase extends THREE.Object3D {
     this.updateVoicer();
   }
   setVoiceEndpoint(voiceId) {
-    if (voiceId) {
-      const url = getVoiceEndpointUrl(voiceId);
+    if (!voiceId) return console.error('voice Id is null')
```
Comment (avaer) [inline character-controller.js:275]: This should throw.

File: avatars/microphone-worklet.js (hunk 1 above)
Comment (lalalune) [inline avatars/microphone-worklet.js:114]: throw

File: character-controller.js [hunk 4]
```diff
@@ -259,16 +272,33 @@ class PlayerBase extends THREE.Object3D {
...
+    if (url) {
       this.voiceEndpoint = new VoiceEndpoint(url);
     } else {
       this.voiceEndpoint = null;
     }
     this.updateVoicer();
   }
   getVoice() {
-    return this.voiceEndpoint || this.voicePack || null;
+    return this.voiceEndpoint || this.voicePack || console.error('no voice endpoint set');
```
Comment (avaer) [inline character-controller.js:301]: If this is an error case, it should throw.

File: character-controller.js [hunk 5]
```diff
@@ -1055,6 +1072,22 @@ class LocalPlayer extends UninterpolatedPlayer {
       }
     });
   }
+  setMicMediaStream(mediaStream) {
+    if (!this.avatar) return console.warn("Can't set mic media stream, no avatar");
```
Comment (avaer) [inline character-controller.js:1076]: This should throw.

File: character-hups.js [hunk 6]
```diff
@@ -66,6 +66,9 @@ class Hup extends EventTarget {
         return this.parent.player.voicer.start(preloadedMessage);
       });
     } else {
+      console.error('No voicer for message', this.fullText);
```
Comment (lalalune) [inline character-hups.js:69]: throw error

File: game.js [hunk 7]
```diff
@@ -1658,7 +1658,12 @@ class GameManager extends EventTarget {
   /* loadVoicePack(voicePack) {
     return localPlayer.loadVoicePack(voicePack);
   } */
+  setVoicePack(voicePack) {
+    const localPlayer = metaversefileApi.useLocalPlayer();
+    return localPlayer.setVoicePack(voicePack);
+  }
   setVoiceEndpoint(voiceId) {
+    if(!voiceId) return console.error('setVoiceEndpoint error', voiceId);
```
Comment (lalalune) [inline game.js:1666]: throw

File: avatars/avatars.js [hunk 8]
```diff
@@ -1973,56 +1974,115 @@ class Avatar {
...
+      this.volume = 0;
+
+      const audioContext = getAudioContext();
+      if (audioContext.state === 'suspended') {
+        (async () => {
+          await audioContext.resume();
+        })();
+      }
+      const _volume = e => {
+        if (!this.manuallySetMouth) {
```
Comment (lalalune) [inline avatars/avatars.js:1998]: add comment -- if the mouth isn't chomping or gulping, basically

Comment (avaer) [inline avatars/avatars.js:1998]: `manuallySetMouth` should be documented when declared. It's for chomping and attack screams.

File: character-controller.js (hunk 4 above)
Comment (lalalune) [inline character-controller.js:301]: throw

File: character-controller.js [hunk 9]
```diff
@@ -251,6 +256,14 @@ class PlayerBase extends THREE.Object3D {
     }
     return false;
   }
+  async setVoicePack({audioUrl, indexUrl}) {
+    const self = this;
+    // this.playersArray.doc.transact(function tx() {
+    const voiceSpec = JSON.stringify({audioUrl, indexUrl, endpointUrl: self.voiceEndpoint ? self.voiceEndpoint.url : ''});
+    self.playerMap.set('voiceSpec', voiceSpec);
+    // });
+    this.loadVoicePack({audioUrl, indexUrl})
```
Comment (avaer) [inline character-controller.js:265]: This breaks the `async` chain. Was that the intent?

**Review by avaer — CHANGES_REQUESTED** [review]
Questions and control flow changes.

File: character-controller.js (hunk 9 above)
Comment (avaer) [reply character-controller.js:265]: It seems either this method should be non-`async`, or we should be `await`ing.

**Pull request merged**